### PR TITLE
Update __init__.py

### PR DESCRIPTION
**Pull request by JJiao**
- [x] add args description
- [x] modify reference input interpolation
- [x] merge in latest changes from https://github.com/JJiao/NiftyPAD
- [x] fix unused `k2` issues (upstream?) @JJiao
- [x] ~rebase/clean & merge~

### CODE REVIEW

File: niftypad/api/__init__.py [hunk 1]
```diff
@@ -17,6 +17,21 @@ def kinetic_model(src, dst=None, params=None, model='srtmb_basis', w=None, r1=0.
...
+      # more args descriptions
+      input_interp_method (str): linear, cubic, exp_1, exp_2, feng_srtm
+      (the interpolation method for getting reference input)
+      w (None or numpy array): weights for weighted model fitting
+      r1 (a value): a pre-chosen value between 0 and 1 for r1, used in srtmb_asl_basis
+      k2p (a value): a pre-chosen value for k2p, in second^-1, used in srtmb_k2p_basis, logan_ref_k2p, mrtm_k2p
+      beta_lim (a list): [beta_min, beta_max] for setting the lower and upper limits of beta values in basis functions,
+       used in srtmb_basis, srtmb_k2p_basis, srtmb_asl_basis
+      n_beta (integer): number of beta values / basis functions, used in srtmb_basis, srtmb_k2p_basis, srtmb_asl_basis
+      linear_phase_start (integer): the second when the linear phase starts, used in logan_ref, logan_ref_k2p, mrtm, mrtm_k2p
```
Comment (FHeeman92) [inline niftypad/api/__init__.py:29]: @JJiao You mean, "start time of linear phase in seconds" correct?

File: niftypad/api/__init__.py [hunk 2]
```diff
@@ -17,6 +17,21 @@ def kinetic_model(src, dst=None, params=None, model='srtmb_basis', w=None, r1=0.
...
+      linear_phase_end (integer): the second when the linear phase ends, used in logan_ref, logan_ref_k2p, mrtm, mrtm_k2p
+      km_outputs (list): e.g. ['R1', 'k2', 'BP'] indicates the kinetic parameters to save
+      thr (a value): threshold value between 0 and 1, to create a mask to choose any voxel,
```
Comment (FHeeman92) [inline niftypad/api/__init__.py:32]: "to create a mask to only select voxels within the brain/disregard activity outside the brain "  (that is the goal, right @JJiao?)

File: niftypad/api/__init__.py [hunk 3]
```diff
@@ -44,7 +59,8 @@ def kinetic_model(src, dst=None, params=None, model='srtmb_basis', w=None, r1=0.
     dt = np.asarray(meta['dt'])
     ref = np.asarray(meta['ref'])
     ref = Ref(ref, dt)
-    ref.interp_1cubic()
+    # change ref interpolation to selected method
+    ref.run_interp(input_interp_method=input_interp_method)
```
Comment (FHeeman92) [inline niftypad/api/__init__.py:63]: Does this come with a list of the available methods?

File: niftypad/api/__init__.py [hunk 4]
```diff
@@ -17,6 +17,21 @@ def kinetic_model(src, dst=None, params=None, model='srtmb_basis', w=None, r1=0.
...
+      input_interp_method (str): linear, cubic, exp_1, exp_2, feng_srtm
+      (the interpolation method for getting reference input)
+      w (None or numpy array): weights for weighted model fitting
+      r1 (a value): a pre-chosen value between 0 and 1 for r1, used in srtmb_asl_basis
+      k2p (a value): a pre-chosen value for k2p, in second^-1, used in srtmb_k2p_basis, logan_ref_k2p, mrtm_k2p
+      beta_lim (a list): [beta_min, beta_max] for setting the lower and upper limits of beta values in basis functions,
+       used in srtmb_basis, srtmb_k2p_basis, srtmb_asl_basis
+      n_beta (integer): number of beta values / basis functions, used in srtmb_basis, srtmb_k2p_basis, srtmb_asl_basis
+      linear_phase_start (integer): the second when the linear phase starts, used in logan_ref, logan_ref_k2p, mrtm, mrtm_k2p
+      linear_phase_end (integer): the second when the linear phase ends, used in logan_ref, logan_ref_k2p, mrtm, mrtm_k2p
```
Comment (FHeeman92) [inline niftypad/api/__init__.py:30]: Same comment applies: "end time of linear phase in seconds" @JJiao

File: niftypad/api/__init__.py (hunk 3 above)
Comment (casperdcl) [reply niftypad/api/__init__.py:63]: I think it's this:

https://github.com/AMYPAD/NiftyPAD/blob/75012a1636bed597ac70e3204faf8e86b969d9bf/niftypad/api/__init__.py#L21

File: niftypad/api/__init__.py (hunk 1 above)
Comment (JJiao) [reply niftypad/api/__init__.py:30]: Yes you are right.

File: niftypad/api/__init__.py (hunk 2 above)
Comment (JJiao) [reply niftypad/api/__init__.py:33]: Not exactly, it only selects the voxels where the activities are higher than a thr related value, it does not necessarily recognise the brain.

File: niftypad/api/__init__.py (hunk 3 above)
Comment (JJiao) [reply niftypad/api/__init__.py:64]: Yes as Casper said.

File: niftypad/api/__init__.py (hunk 4 above)
Comment (JJiao) [reply niftypad/api/__init__.py:31]: Yes that's right.

File: niftypad/models.py [hunk 5]
```diff
@@ -271,29 +350,360 @@
...
+    # find tt for the linear phase
+    tt = np.logical_and(mft >= linear_phase_start, mft <= linear_phase_end)
     tt = tt[1:]
+
     mft = mft[1:]
     reg = LinearRegression(fit_intercept=False).fit(xx[tt,], yy[tt])
     bp = -reg.coef_[0] / reg.coef_[1] - 1
+
+    # for 1 TC
+    k2 = -reg.coef_[1]
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply niftypad/models.py:381]: @JJiao I merged in your latest changes from https://github.com/JJiao/NiftyPAD and it seems there's quite a few cases of `k2` being defined but not used.

[the remark this answers is not in the document]
Comment (casperdcl) [reply niftypad/models.py:381]: @JJiao not sure why this was markes as resolved - it's still a problem

**Pull request merged**